### PR TITLE
feat: Add skeleton for api key module

### PR DESCRIPTION
**Pull request by sradevski**
Adds a module skeleton for the API Key module.
Implementation of functionalities will follow in separate PRs

### CODE REVIEW

File: packages/api-key/integration-tests/__tests__/api-key-module-service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Modules } from "@medusajs/modules-sdk"
+import { initModules } from "medusa-test-utils/dist"
+import { IApiKeyModuleService } from "@medusajs/types"
+import { getInitModuleConfig, MikroOrmWrapper } from "../utils"
+
+describe("api key module service", function () {
```
Comment (adrien2p) [inline packages/api-key/integration-tests/__tests__/api-key-module-service.spec.ts]: todo: use the new util `moduleIntegrationTestRunner`

File: packages/api-key/integration-tests/setup-env.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if (typeof process.env.DB_TEMP_NAME === "undefined") {
```
Comment (adrien2p) [inline packages/api-key/integration-tests/setup-env.js]: todo: rm file

File: packages/api-key/integration-tests/setup.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { JestUtils } from "medusa-test-utils"
```
Comment (adrien2p) [inline packages/api-key/integration-tests/setup.js]: todo: rm file

File: packages/api-key/integration-tests/utils/database.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { TestDatabaseUtils } from "medusa-test-utils"
```
Comment (adrien2p) [inline packages/api-key/integration-tests/utils/database.ts]: todo: rm file

File: packages/api-key/integration-tests/utils/get-init-module-config.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Modules, ModulesDefinition } from "@medusajs/modules-sdk"
```
Comment (adrien2p) [inline packages/api-key/integration-tests/utils/get-init-module-config.ts]: todo: rm file

File: packages/api-key/integration-tests/utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export * from "./database"
```
Comment (adrien2p) [inline packages/api-key/integration-tests/utils/index.ts]: todo: rm file

File: packages/api-key/jest.config.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        tsConfig: "tsconfig.spec.json",
+        isolatedModules: true,
+      },
+    ],
+  },
+  testEnvironment: `node`,
+  moduleFileExtensions: [`js`, `ts`],
+  modulePathIgnorePatterns: ["dist/"],
+  setupFiles: ["<rootDir>/integration-tests/setup-env.js"],
+  setupFilesAfterEnv: ["<rootDir>/integration-tests/setup.js"],
```
Comment (adrien2p) [inline packages/api-key/jest.config.js]: todo: rm

File: packages/api-key/src/services/api-key-module-service.ts [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext,
+  ModulesSdkUtils,
+} from "@medusajs/utils"
+
+import { entityNameToLinkableKeysMap, joinerConfig } from "../joiner-config"
+import { ApiKey } from "@models"
+
+const generateMethodForModels = [ApiKey]
```
Comment (adrien2p) [inline packages/api-key/src/services/api-key-module-service.ts]: todo: let it empty

**Review by adrien2p — APPROVED** [review]
LGTM 🔥 , left a couple of todo's feel free to change them in a follow up pr if you prefer

File: packages/types/src/api-key/service.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+   * @param sharedContext
+   */
+  revoke(id: string, sharedContext?: Context): Promise<void>
+
+  /**
+   * Check the validity of an api key
+   * @param id
+   * @param sharedContext
+   */
+  authenticate(id: string, sharedContext?: Context): Promise<boolean>
```
Comment (adrien2p) [inline packages/types/src/api-key/service.ts:80]: q: rm authenticate? or is the naming expected?

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**